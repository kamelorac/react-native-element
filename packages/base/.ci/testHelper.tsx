import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
