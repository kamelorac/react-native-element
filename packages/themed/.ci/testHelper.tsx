import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';
import {
  ThemeProvider,
  FullTheme,
  createTheme,
  CreateThemeOptions,
} from '../src/config';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  themeProp: Partial<CreateThemeOptions> = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(
    <ThemeProvider theme={createTheme(themeProp)}>{children}</ThemeProvider>,
    options
  );
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
