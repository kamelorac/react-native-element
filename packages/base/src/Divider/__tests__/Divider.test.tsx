import React from 'react';
import { Divider } from '../index';
import { renderWithWrapper } from '../../../.ci/testHelper';

describe('Divider Component', () => {
  it('should match snapshot', () => {
    const component = renderWithWrapper(<Divider />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render with style', () => {
    const { wrapper } = renderWithWrapper(
      <Divider style={{ borderLeftColor: 'blue' }} />,
      'RNE__Divider'
    );
    expect(wrapper.props.style).toMatchObject({
      borderLeftColor: 'blue',
    });
  });

  it('should render with color and orientation horizontal', () => {
    const { wrapper } = renderWithWrapper(
      <Divider color={'red'} />,
      'RNE__Divider'
    );
    expect(wrapper.props.style).toMatchObject({
      borderBottomColor: 'red',
    });
  });

  it('should render with color and orientation vertical', () => {
    const { wrapper } = renderWithWrapper(
      <Divider color={'pink'} orientation="vertical" />,
      'RNE__Divider'
    );
    expect(wrapper.props.style).toMatchObject({
      borderRightColor: 'pink',
    });
  });

  it('should render with inset true and insetType left', () => {
    const { wrapper } = renderWithWrapper(
      <Divider inset insetType="left" />,
      'RNE__Divider'
    );
    expect(wrapper.props.style).toMatchObject({
      marginLeft: 72,
    });
  });

  it('should render with inset true and insetType right', () => {
    const { wrapper } = renderWithWrapper(
      <Divider inset={true} insetType="right" />,
      'RNE__Divider'
    );
    expect(wrapper.props.style).toMatchObject({
      marginRight: 72,
    });
  });

  it('should render with orientation vertical', () => {
    const { wrapper } = renderWithWrapper(
      <Divider orientation="vertical" />,
      'RNE__Divider'
    );
    expect(wrapper.props.style).toMatchObject({
      height: 'auto',
      alignSelf: 'stretch',
    });
  });

  it('should render with width and orientation horizontal', () => {
    const { wrapper } = renderWithWrapper(
      <Divider width={5} />,
      'RNE__Divider'
    );
    expect(wrapper.props.style).toMatchObject({
      borderBottomWidth: 5,
    });
  });

  it('should render with subHeader and subHeaderStyle', () => {
    const component = renderWithWrapper(
      <Divider subHeader="Test" subHeaderStyle={{ color: 'blue' }} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render with subHeader and subHeaderStyle and inset true', () => {
    const component = renderWithWrapper(
      <Divider
        subHeader="Test"
        inset={true}
        subHeaderStyle={{ color: 'blue' }}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
