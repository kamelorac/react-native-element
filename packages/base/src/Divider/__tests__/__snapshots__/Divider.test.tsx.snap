// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      {
        "borderBottomColor": "#bcbbc1",
        "borderBottomWidth": 0.5,
      }
    }
    testID="RNE__Divider"
  />
</View>
`;

exports[`Divider Component should render with subHeader and subHeaderStyle 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      {
        "borderBottomColor": "#bcbbc1",
        "borderBottomWidth": 0.5,
      }
    }
    testID="RNE__Divider"
  />
  <Text
    style={
      {
        "color": "blue",
        "includeFontPadding": false,
      }
    }
  >
    Test
  </Text>
</View>
`;

exports[`Divider Component should render with subHeader and subHeaderStyle and inset true 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      {
        "borderBottomColor": "#bcbbc1",
        "borderBottomWidth": 0.5,
        "marginLeft": 72,
      }
    }
    testID="RNE__Divider"
  />
  <Text
    style={
      {
        "color": "blue",
        "includeFontPadding": false,
        "marginLeft": 72,
      }
    }
  >
    Test
  </Text>
</View>
`;
